src/extra/Markdown/plugins/decorator.ts

```diff
@@ -0,0 +1,106 @@
+import unified from 'unified';
+import visit from 'unist-util-visit';
+
+const splitTextIntoNodes = (text: string, decorator: Decorator) => {
+	// Reset the regex index state
+	decorator.match.lastIndex = 0;
+
+	const nodes: any[] = [];
+	let lastIndex = 0;
+
+	while (true) {
```

Comment:
Instead of keeping indices in the decorator, how about doing:
```
const nodes = [];
let lastIndex = 0;
while ((match = re.exec(str)) != null) {
  const matchedString = match[0];
  nodes.push(
    {
      data: str.slice(lastIndex, match.index)
    },
    {
      data: matchedString,
    }
  )

  lastIndex = match.index + matchedString.length
}

if(str.length > lastIndex){
  nodes.push({
    data: str.slice(lastIndex)
  })
}

```

Reply:
@sradevski it's not me keeping indices in the decorator. `RegExp` becomes stateful when it has `g` (global) flag. So when doing `regex.ecec`, it modifies `regex.lastIndex`. Also linter does not allow `while ((match = re.exec(str)) != null) {`.